agentendpoint/config_task.go

```diff
@@ -328,22 +342,38 @@ func (c *configTask) run(ctx context.Context) error {
 				break
 			}
 			if enforcementActionTaken := enforceConfigResourceState(ctx, plcy, rCompliance, configResource); enforcementActionTaken {
-				// On any change we trigger post check.
-				c.postCheckRequired = true
+				// On any change we trigger post check for all previous resouces.
+				c.markPostCheckRequired()
 			}
 			if plcy.hasError {
 				break
 			}
 		}
 	}
 
-	if c.postCheckRequired {
-		c.postCheckState(ctx)
-	}
+	// Run any post checks that we need to.
+	c.postCheckState(ctx)
 
 	return c.reportCompletedState(ctx, "", agentendpointpb.ApplyConfigTaskOutput_SUCCEEDED)
 }
 
+// Mark all resources that have already completed as "needs post check".
+func (c *configTask) markPostCheckRequired() {
+	for _, osPolicy := range c.Task.GetOsPolicies() {
+		plcy, ok := c.policies[osPolicy.GetId()]
+		if !ok || plcy.hasError {
+			continue
+		}
+		for _, configResource := range osPolicy.GetResources() {
+			res, ok := plcy.resources[configResource.GetId()]
+			if !ok || res == nil {
```

Comment:
When would it be nil?

Reply:
With the current logic never (production code we never set this to nil), this was a helper for unit tests where we can run these functions out of order and/or pre-populate fields with empty values.